### ISSUE TITLE
NativeAOT: System.Reactive causes ilc fail to generate code in Release configuration

### ISSUE THREAD

**Issue by hez2010** [dotnet/runtimelab#534]
In ilc 6.0.0-alpha.1.21055.3, `System.Reactive` causes ilc fail to generate code while using `Release` configuration:

```bash
dotnet publish -c Release -r win-x64
```

However, when use Debug configuration (`-c Debug`), ilc only reports a warning about `CatchScheduler` and the code can be compiled without exceptions.

Exception stacktrace:

```
  Unhandled exception. ILCompiler.CodeGenerationFailedException: Code generation failed for method '[System.Reactive]System.Reactive.Concurrency.CatchScheduler`1+CatchSchedulerPeriodic+PeriodicallyScheduledWorkItem`1<System.__Canon,System.ValueTuple`2<System.__Canon,System.__Canon>>.Tick(ValueTuple`2<__Canon,__Canon>)'
   ---> System.InvalidOperationException: [System.Reactive]System.Reactive.Concurrency.CatchScheduler`1+CatchSchedulerPeriodic+PeriodicallyScheduledWorkItem`1<System.__Canon,System.ValueTuple`2<System.__Canon,System.__Canon>>: TypeHandle: [S.P.CoreLib]System.ValueTuple`2<System.Reactive.Concurrency.CatchScheduler`1+CatchSchedulerPeriodic+PeriodicallyScheduledWorkItem`1<TException_System.__Canon,TState_System.ValueTuple`2<System.__Canon,System.__Canon>>,TState_System.ValueTuple`2<System.__Canon,System.__Canon>>
     at ILCompiler.DependencyAnalysis.PrecomputedDictionaryLayoutNode.GetSlotForEntry(GenericLookupResult entry)
     at ILCompiler.Compilation.ComputeGenericLookup(MethodDesc contextMethod, ReadyToRunHelperId lookupKind, Object targetOfLookup)
     at Internal.JitInterface.CorInfoImpl.ComputeLookup(CORINFO_RESOLVED_TOKEN& pResolvedToken, Object entity, ReadyToRunHelperId helperId, CORINFO_LOOKUP& lookup)
     at Internal.JitInterface.CorInfoImpl.embedGenericHandle(CORINFO_RESOLVED_TOKEN& pResolvedToken, Boolean fEmbedParent, CORINFO_GENERICHANDLE_RESULT& pResult)
     at Internal.JitInterface.CorInfoImpl._embedGenericHandle(IntPtr thisHandle, IntPtr* ppException, CORINFO_RESOLVED_TOKEN* pResolvedToken, Byte fEmbedParent, CORINFO_GENERICHANDLE_RESULT* pResult)
     --- End of inner exception stack trace ---
     at Internal.JitInterface.CorInfoImpl.CompileMethodInternal(IMethodNode methodCodeNodeNeedingCode, MethodIL methodIL)
     at Internal.JitInterface.CorInfoImpl.CompileMethod(MethodCodeNode methodCodeNodeNeedingCode, MethodIL methodIL)
     at ILCompiler.RyuJitCompilation.CompileSingleMethod(CorInfoImpl corInfo, MethodCodeNode methodCodeNodeNeedingCode)
     at System.Threading.ThreadPoolWorkQueue.Dispatch()
```

This prevents Avalonia app being native aot compiled.

**Comment by MichalStrehovsky** [comment]
Thanks! I have a fix in #536.

**Comment by MichalStrehovsky** [comment]
Fixed in #536.

**Comment by hez2010** [comment]
Thanks!
